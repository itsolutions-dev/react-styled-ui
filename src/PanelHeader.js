import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
  getTextColor,
  getBackgroundColor,
} from './utils/';

const PanelHeader = styled({
  component: 'div',
  style: css`
    font-size: 24px;
    font-weight: 600;
    padding: 20px;
    color: ${props => getTextColor(props)};
    ${(props) => {
      const color = props.borderColor !== undefined ? props.borderColor : getBackgroundColor(props);
      return `
        background: ${color};
        border: 1px solid ${color};
      `;
    }}
    border-top-left-radius: ${props => getSizeOrDefault(props.radius, props.theme.radius)};
    border-top-right-radius: ${props => getSizeOrDefault(props.radius, props.theme.radius)};
  `,
});

PanelHeader.defaultProps = {
  primary: true,
};

export default PanelHeader;
