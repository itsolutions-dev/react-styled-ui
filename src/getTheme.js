import defaultTheme from './theme';
import deepMerge from './utils/deepMerge';

const getTheme = deepMerge.bind(null, defaultTheme);

export default getTheme;
