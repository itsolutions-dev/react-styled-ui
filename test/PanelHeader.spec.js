import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import getTheme from '../src/getTheme';
import PanelHeader from '../src/PanelHeader';

const theme = getTheme({
  radius: '8px',
  palette: {
    primary: 'blue',
    primaryText: 'white',
  },
});

describe('PanelHeader', () => {
  test('should render a PanelHeader', () => {
    const component = ReactTestRenderer.create(<PanelHeader theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<PanelHeader theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have the given border color', () => {
    const component = ReactTestRenderer.create(<PanelHeader borderColor="green" theme={theme} />);
    expect(component).toHaveStyleRule('background', 'green');
    expect(component).toHaveStyleRule('border', '1px solid green');
  });

  test('should be primary by default', () => {
    const component = ReactTestRenderer.create(<PanelHeader theme={theme} />);
    expect(component).toHaveStyleRule('color', 'white');
    expect(component).toHaveStyleRule('background', 'blue');
    expect(component).toHaveStyleRule('border', '1px solid blue');
  });

  test('should have the given radius', () => {
    const component = ReactTestRenderer.create(<PanelHeader radius={10} theme={theme} />);
    expect(component).toHaveStyleRule('border-top-left-radius', '10px');
    expect(component).toHaveStyleRule('border-top-right-radius', '10px');
  });

  test('should get radius from theme', () => {
    const component = ReactTestRenderer.create(<PanelHeader theme={theme} />);
    expect(component).toHaveStyleRule('border-top-left-radius', '8px');
    expect(component).toHaveStyleRule('border-top-right-radius', '8px');
  });
});
