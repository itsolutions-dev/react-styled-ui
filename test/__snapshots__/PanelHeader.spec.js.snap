// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelHeader should render a PanelHeader 1`] = `
.c0 {
  font-size: 24px;
  font-weight: 600;
  padding: 20px;
  color: white;
  background: blue;
  border: 1px solid blue;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
}

<div
  className="c0"
/>
`;
